pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -224,6 +151,269 @@ func (a *AmbientIndex) matchesScope(scope model.WaypointScope, w *model.Workload
 	return true
 }
 
+func PickBestPod(pods []*v1.Pod) *v1.Pod {
+	var best *v1.Pod
+	for _, p := range pods {
+		if !IsPodRunning(p) {
+			continue
+		}
+		if best == nil || p.CreationTimestamp.After(best.CreationTimestamp.Time) {
+			best = p
+		}
+	}
+	return best
+}
+
+func (a *AmbientIndex) Recompute(name types.NamespacedName) *model.WorkloadInfo {
+	ip := name.Name
+	log := log.WithLabels("controller", "ambient", "ip", ip)
+	log.Debugf("reconcile")
+
+	// Should be 0 or 1 pod, unless we are in a weird state
+	pod := PickBestPod(a.podIPIndex.Lookup(ip))
+
+	// Should be 0 or 1 entry, unless we are in a weird state
+	var entry *networking.WorkloadEntry
+	var configEntry *config.Config
+	if a.workloadEntriesIPIndex != nil {
+		entries := a.workloadEntriesIPIndex.Lookup(ip)
+		if len(entries) > 0 {
+			configEntry = entries[0]
+			entry = serviceentry.ConvertWorkloadEntry(*entries[0])
+		}
```

Comment:
is this lookup deterministic? i.e. are the entries sorted when they are returned?

Reply:
the sorting on endpoints using `Lookup()` below suggests to me that it's not sorted